pydantic/main.py

```diff
@@ -1094,18 +1095,33 @@ def _calculate_keys(self, *args: Any, **kwargs: Any) -> Any:
 RootModelRootType = typing.TypeVar('RootModelRootType')
 
 
+class _RootModelSpecialValues(enum.Enum):
+    no_value = enum.auto()
+
+
+_RootModelNoValue = _RootModelSpecialValues.no_value
```

Comment:
Could we just do `_RootModelNoValue = object()` and not introduce an enum for this? Or at least do the same thing we do with `_Undefined`:

https://github.com/pydantic/pydantic/blob/cf48b1a5a494604ec161ed85df6285d683084baa/pydantic/_internal/_fields.py#L39C7-L57

(We could also just use `pydantic._internal._fields.Undefined` for this directly?)

Reply:
I like the idea to just use the same `Undefined` here. Thank you!